### PR TITLE
chore(go): update enum casing

### PR DESCRIPTION
**Pull request by Fluf22**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1641]

### Changes included:

- Make enum have a upper snake case

[DI-1641]: https://algolia.atlassian.net/browse/DI-1641?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java [hunk 1]
```diff
@@ -29,6 +30,10 @@ public void processOpts() {
     additionalProperties.put("packageName", client.equals("query-suggestions") ? "suggestions" : client);
     additionalProperties.put("enumClassPrefix", true);
     additionalProperties.put("isSearchClient", client.equals("search"));
+    additionalProperties.put(
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java:33]: the preferred way is to use `addMustacheLambdas` like [here](https://github.com/algolia/api-clients-automation/blob/main/generators/src/main/java/com/algolia/codegen/cts/AlgoliaCTSGenerator.java#L92) but that works too

File: generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java [hunk 2]
```diff
@@ -29,6 +30,10 @@ public void processOpts() {
...
+      "lambda.toUpperSnakeCase",
```
Comment (millotp) [inline generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java:34]: I think the official name is `SCREAMING_SNAKE_CASE` ahah

**Review by millotp — COMMENTED** [review]
the DX is way better, thanks !

File: generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java (hunk 2 above)
Comment (Fluf22) [reply generators/src/main/java/com/algolia/codegen/AlgoliaGoGenerator.java:34]: I'll update it, I was wondering what naming to use

**Comment by Fluf22** [comment]
The rebase + conflict resolution was failing because JetBrains kept updating the go.mod+sum of the playground, which made me do the edit manually, and I must have missed something since some changes were reverted
I'll update again and push a fix

**Review by millotp — APPROVED** [review]
beautiful !

**Pull request merged**